### PR TITLE
anki: more no-op exp cleanup

### PR DESCRIPTION
**Pull request by bhipple**
Continued progress towards #89745; just simplifying a bit.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/anki/default.nix [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 let
   # when updating, also update rev-manual to a recent version of
-  # https://github.com/dae/ankidocs
+  # https://github.com/ankitects/ankidocs
```
Comment (Profpatsch) [inline pkgs/games/anki/default.nix:34]: dead link

File: pkgs/games/anki/default.nix [hunk 2]
```diff
@@ -42,7 +42,7 @@ let
     pname = "anki-manual";
     inherit version;
     src = fetchFromGitHub {
-      owner = "dae";
+      owner = "ankitects";
```
Comment (Profpatsch) [inline pkgs/games/anki/default.nix:45]: That’s not a no-op holy <s>f</s> heck.

I do hope that slipped into this commit by mistake.

File: pkgs/games/anki/default.nix [hunk 3]
```diff
@@ -112,16 +112,10 @@ buildPythonApplication rec {
...
-    # Dummy build phase
-    # Anki does not use setup.py
-  '';
+  # Anki does not use setup.py
+  dontBuild = true;
 
   postPatch = ''
-    # Remove unused starter. We'll create our own, minimalistic,
-    # starter.
-    # rm anki/anki
```
Comment (Profpatsch) [inline pkgs/games/anki/default.nix:123]: Is this not relevant anymore?

**Review by Profpatsch — CHANGES_REQUESTED** [review]

File: pkgs/games/anki/default.nix (hunk 2 above)
Comment (Ekleog) [reply pkgs/games/anki/default.nix:45]: This being said, it looks like ankisrs.net now points to the ankitects repo, so IMO this is not a problem (even though it doesn't match the commit name). (Also, the build will probably not be affected, as the reference and hash stay the same)

File: pkgs/games/anki/default.nix (hunk 1 above)
Comment (bhipple) [reply pkgs/games/anki/default.nix:34]: Oops, it's https://github.com/ankitects/anki-docs now, with a dash on the RHS.

File: pkgs/games/anki/default.nix (hunk 2 above)
Comment (bhipple) [reply pkgs/games/anki/default.nix:45]: https://github.com/dae/ankidocs is a URL redirect to https://github.com/ankitects/anki-docs now, so it is a no-op.  I can change the `sha256` to garbage and nix will re-fetch and tell me the same hash.  Am I missing something?

File: pkgs/games/anki/default.nix (hunk 3 above)
Comment (bhipple) [reply pkgs/games/anki/default.nix:123]: It was commented out back in 2018 and we seem to have been fine without it, so I'd say so:
https://github.com/NixOS/nixpkgs/commit/a30e30432da95d6eae8713b57ca5708d892a3263

File: pkgs/games/anki/default.nix (hunk 2 above)
Comment (Profpatsch) [reply pkgs/games/anki/default.nix:45]: Yeah, please put it in a different commit explaining that.

Comment (bhipple) [reply pkgs/games/anki/default.nix:45]: Done

**Pull request merged**